packages/blockchain-wallet-v4/src/redux/payment/bch/sagas.js

```diff
@@ -269,11 +270,24 @@ export default ({ api }) => {
   }
 
   // ///////////////////////////////////////////////////////////////////////////
-  const calculatePublish = function*(txHex) {
+  const calculatePublish = function*(txHex, lockSecret) {
     if (!txHex) {
       throw new Error('missing_signed_tx')
     }
-    return yield call(() => taskToPromise(pushBitcoinTx(txHex)))
+    return yield call(() => taskToPromise(pushBitcoinTx(txHex, lockSecret)))
+  }
+
+  // ///////////////////////////////////////////////////////////////////////////
+  const splitCoins = function*(selection) {
+    const dust = yield call(api.getBchDust)
+    const lockSecret = prop('lock_secret', dust)
+    const address = scriptToAddress(prop('output_script', dust))
+    const coinDust = Coin.fromJS({ address, value: 546 })
+
+    selection.inputs.push(coinDust)
+    selection.outputs.push(coinDust)
```

Comment:
here you are mutating the selection (which is the argument that you received). I don't know how this can effect outside of this function so I would not do that...